### PR TITLE
cmsis-dsp: support native posix

### PR DESCRIPTION
**Pull request by shlomow**

### CODE REVIEW

**Review by galak — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
On second thought, will you please update the testcase.yaml files in `tests/lib/cmsis_dsp` to make sure we exercise this in CI?

**Comment by nashif** [comment]
Will it be enough to just test this on native_posix in CI, i.e.:

```
+++ b/tests/lib/cmsis_dsp/basicmath/testcase.yaml
@@ -2,9 +2,7 @@ tests:
   libraries.cmsis_dsp.basicmath:
     filter: (CONFIG_CPU_CORTEX_R or CONFIG_CPU_CORTEX_M) and TOOLCHAIN_HAS_NEWLIB == 1
     integration_platforms:
-      - frdm_k64f
-      - sam_e70_xplained
-      - mps2_an521
+      - native_posix
     tags: cmsis_dsp
     min_flash: 128
     min_ram: 64
```

**Comment by shlomow** [comment]
@nashif I think that at least one arm platform should be tested, since cmsis-dsp compiles differently in that case. What do you think?

**Comment by stephanosio** [comment]
> Will it be enough to just test this on native_posix in CI, i.e.:

@nashif Noting that the CMSIS-DSP component is intended to be used on the ARM platforms, it would be beneficial to keep those in the list:

* frdm_k64f - ARMv7 / Cortex-M4
* sam_e70_xplained - ARMv7 / Cortex-M7
* mps2_an521 - ARMv8 / Cortex-M33

**Review by stephanosio — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
Thank you!

File: west.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ manifest:
   # Please add items below based on alphabetical order
   projects:
     - name: cmsis
-      revision: 421dcf358fa420e9721a8452c647f0d42af8d68c
+      revision: 5d18367d9425a278c5931ed7dc0c4491e98150b1
```
Comment (galak) [inline west.yml:29]: Can we set this to `revision: pull/10/head` to get testing of that PR.

Comment (shlomow) [reply west.yml:29]: Thanks! I was searching this now :)

**Review by galak — APPROVED** [review]

**Pull request merged**